server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -218,17 +218,20 @@ public Rounding build() {
     static class TimeUnitRounding extends Rounding {
 
         static final byte ID = 1;
+        static final long TZ_OFFSET_NON_FIXED = -1;
 
         private final DateTimeUnit unit;
         private final ZoneId timeZone;
         private final boolean unitRoundsToMidnight;
-        private final boolean isUtcTimeZone;
+        /** For fixed offset timezones, this is the offset in milliseconds, otherwise TZ_OFFSET_NON_FIXED */
+        private final long fixedOffset;
```

Comment:
ignoreable nit: maybe name this fixedOffsetMs or something to indicate its contents